dart/test/sentry_test.dart

```diff
@@ -157,6 +157,16 @@ void main() {
           throwsArgumentError);
     },
   );
+
+  test('options.environment not null', () async {
+    await Sentry.init(
+      (options) {
+        options.dsn = fakeDsn;
+        expect(options.environment, isNotNull);
+      },
+      appRunner,
+    );
+  });
```

Comment:
the test makes sense, but I'd actually prefer to have specific unit tests asserting that its `debug` or `profile`, can we do that?

`isReleaseMode` and `isProfileMode` as it is probably will difficult the tests as they are `const` and not mockable, we could create a typedef function instead, so we can mock/replace it for testing.

Reply:
btw for doing this, you might need:

https://github.com/getsentry/sentry-dart/projects/1#card-50187067

Reply:
Ah yes, having an interface and replacing the implementation with a mock for unit tests would be neat. I'm not aware of how to run tests with release/profile modes, maybe https://dart.dev/tools/dart2native would be of help. But the cleanest way would be what you are suggesting.

Reply:
Should we wait for the other task and leave this PR open in the meantime? Think this would be the cleanest solution to test all possibilities.

Reply:
@denrase I think it'd be fine to start working on a half solution for the PlatformChecker interface, but you only add the necessary methods for this PR which is probably IsReleaseMode(), isDebugMode, isProfileMode, and for the other task, you add the next few methods, so we could test this PR and merge it, wdyt?

Reply:
Yeah sounds good, will do!

Reply:
Went with a class for `PlatformChecker`, as we have implicit interfaces in dart. Not too sure about the placement of the const instance for the init default. Would love to hear your feedback. 🙇